crates/hdi/src/x_salsa20_poly1305.rs

```diff
@@ -44,3 +44,22 @@ pub fn x_25519_x_salsa20_poly1305_decrypt(
             ))
     })
 }
+
+/// Libsodium crypto_box decryption, but converts ed25519 *signing*
+/// keys into x25519 encryption keys.
+/// WARNING: Please first understand the downsides of using this function:
```

Comment:
Does this warning primarily apply if you're passing a key you're also using as an AgentPubKey or is it just as relevant if you've created a new signing keypair and are choosing to use that for encryption/decryption? 

Maybe the `AgentPubKey` parameter answers that question

Reply:
The warning applies to doing encryption with signing keys in general. I don't think there's a use-case in holochain for creating your own signing keys and doing encryption with them... because you could just create encryption keys instead. The only reason we're allowing this is because the signing keys are already widely published, so it makes it easy to encrypt something for a specific target.

Technically this api doesn't disallow that, lair doesn't know about agents or not... so you can create the AgentPubKey struct out of raw signing key bytes and this function will still work.

Reply:
Got it thanks, that's clearer to me now